Remove confirm keyword requirement from device reboot

mundruid opened the pull request:
- Remove confirm keyword from all devices reboot
- Addresses issue #94

jmcgill298 reviewed (changes requested):
Thanks @mundruid, one comment, can you add `**kwargs` to the reboot method and send a warning that passing `confirm` is deprecated if confirm is in kwargs?

```
def reboot(..., **kwargs):
    if confirm in reboot:
        warnings.warn("passing 'confirm' to reboot method is deprecated..", DeprecationWarning)
```

mundruid commented:
> Thanks @mundruid, one comment, can you add `**kwargs` to the reboot method and send a warning that passing `confirm` is deprecated if confirm is in kwargs?
> 
> ```
> def reboot(..., **kwargs):
>     if confirm in reboot:
>         warnings.warn("passing 'confirm' to reboot method is deprecated..", DeprecationWarning)
> ```

Thanks for the review @jmcgill298 . It's a good point for backward compatibility and I added this. 

I added tests in the unit tests to check if the warning is triggered. We can remove them if we consider them unnecessary. I only added unit tests for two types unit with fixtures and without, i.e., asa and aireos, since the deprecation code is the same for all.

jmcgill298 commented on pyntc/devices/asa_device.py:
```diff
@@ -379,31 +379,44 @@ def peer_redundancy_state(self):
-            signal.signal(signal.SIGALRM, handler)
-            signal.alarm(10)
+        Raises:
+            RebootSignal: When the device is still unreachable after the timeout period.
```
```suggestion
            RebootTimeoutError: When the device is still unreachable after the timeout period.
```

jmcgill298 commented:
Thanks @mundruid I rebased b/c I saw commits that shouldn't be there, then I realized that the issue was the PR was into the main branch instead of develop. I fixed that, and I updated an incorrect docstring (it wasn't something you changed, just noticed it in the review). I will merge once tests pass
